### ISSUE TITLE
`initialViewState` vs `viewState` on `<DeckGL />`

### ISSUE THREAD

**Issue by alexung** [uber/deck.gl#3867]
#### Description
Our `<DeckGL />` component looks like:

```
<DeckGL
     layers={layers}
     controller={true}
     initialViewState={mapStore.viewport}
     viewState={mapStore.viewport}
     onViewStateChange={handleViewStateChange}
     >...</DeckGL>
```
We have some pin dragging functionality on that previously worked on DeckGL v6.4.0, which essentially locks `onViewStateChange` when you've clicked a pin but, after upgrading to v7.2.6, it seems that when dragging pins across the map would not lock the viewport anymore and the dragging functionality was broken.

We realized that perhaps having both [`initialViewState`](https://github.com/uber/deck.gl/blob/master/docs/api-reference/deck.md#initialviewstate-object) and [`viewState`](https://github.com/uber/deck.gl/blob/master/docs/api-reference/deck.md#viewstate-object) props on the component were causing the bug, and scrapped the [`initialViewState`](https://github.com/uber/deck.gl/blob/master/docs/api-reference/deck.md#initialviewstate-object) prop which succeeded in fixing the dragging capability.  Wanted to understand this, however, as [`initialViewState`](https://github.com/uber/deck.gl/blob/master/docs/api-reference/deck.md#initialviewstate-object) seems that if [`viewState`](https://github.com/uber/deck.gl/blob/master/docs/api-reference/deck.md#viewstate-object) is provided, then it should shadow the [`viewState`](https://github.com/uber/deck.gl/blob/master/docs/api-reference/deck.md#viewstate-object), rather than conflict with it.

I've looked through and linked documentation for both props above without mention of potential clashing between these props and would be appreciative of clarification of this.  Thanks!

#### Environment
 - Framework Version: deck.gl 7.2.6
 - Browser Version: Chrome Version 78.0.3904.87 (Official Build) (64-bit)
 - OS: Mac OS X Mojave 10.14.6

**Comment by Pessimistress** [comment]
Thanks for reporting. If I recall correctly, in 6.x we used to call `setState` when the internal view state changes (turned on by supplying `initialViewState`). In 7.0, we made a performance update that skips the React component update. Since your `viewState` prop does not change during dragging, the internal state change would indeed override the external `viewState`.

I'm glad you are able to work around this issue. I think we should

1. Avoid mutating internal view state without checking `props.viewState`
2. Issue a warning if both `initialViewState` and `viewState` are specified.
